### PR TITLE
useClickOutside hook

### PR DESCRIPTION
**Pull request by imevanc**
The useClickOutside hook detects, if a click event happened outside of a component.
I use it for autosaving a note when the user clicks outside of the input tag.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -470,6 +472,83 @@ None : This hooks takes no parameters.
 
 </br>
 
+## 👆 useClickOutside
```
Comment (aromalanil) [inline docs/README.md:475]: @imevanc Please add a simple example, just include the demonstration of the hook, nothing more

File: hooks/useClickOutside.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ *
+ * Custom hook which return an object with the refence to the DOM element,
+ * the state and the setter of the state.
+ * @param {boolean} initialValue Initial value of the state.
+ * @return {ElementRef} Object containing the coordinates of the mouse pointer.
+ */
+const useClickOutside = (initialValue) => {
+  const [wasClicked, setWasClicked] = React.useState(initialValue);
```
Comment (aromalanil) [inline hooks/useClickOutside.js:17]:
```suggestion
  const [wasClicked, setWasClicked] = React.useState(false);
```

I suggest keeping this false. I don't see a use case for accepting an initial value

File: hooks/useClickOutside.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  const myRef = React.useRef(null);
+
+  const clickOutside = (event) => {
+    if (myRef.current && !myRef.current.contains(event.target)) {
+      setWasClicked(false);
```
Comment (aromalanil) [inline hooks/useClickOutside.js:22]:
```suggestion
      setWasClicked(true);
```

Shouldn't this be `true` as in a click on outside the element was done, like ` wasClicked = true` ?

File: hooks/useClickOutside.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  };
+
+  React.useEffect(() => {
+    // Event listener to track the component activity.
+    document.addEventListener('click', clickOutside, true);
```
Comment (aromalanil) [inline hooks/useClickOutside.js:28]: Is there any specific reason in keeping `useCapture` as true?

**Comment by imevanc** [comment]
@aromalanil , I have completed the suggested changes and resolved the file conflict by accepting the incoming changes from the master branch 😀

**Comment by aromalanil** [comment]
@imevanc  Please resolve the linting issues

**Pull request merged**